web/app/services/session.ts

```diff
@@ -13,17 +13,20 @@ export default class SessionService extends EmberSimpleAuthSessionService {
   // Because we redirect as part of the authentication flow, the parameter storing the transition gets reset. Instead, we keep track of the redirectTarget in browser sessionStorage and override the handleAuthentication method as recommended by ember-simple-auth.
 
   handleAuthentication(routeAfterAuthentication: string) {
-    let redirectObject = window.localStorage.getItem(
-      REDIRECT_LOCAL_STORAGE_KEY
-    );
+    let redirectStorageValue =
+      window.sessionStorage.getItem(REDIRECT_STORAGE_KEY) ||
+      window.localStorage.getItem(REDIRECT_STORAGE_KEY);
 
     let redirectTarget: string | null = null;
     let transition;
 
-    if (redirectObject) {
-      // Check if the object is less than 2 minutes old
-      if (Date.now() < JSON.parse(redirectObject).expiresOn) {
-        redirectTarget = JSON.parse(redirectObject).url;
+    if (redirectStorageValue) {
+      const isObject = redirectStorageValue.startsWith("{");
```

Comment:
Is this an idiomatic way to test for this, or would it maybe be better to use `typeof` or something?

Reply:
`typeof` was also my first thought, but all localStorage/sessionStorage values are saved as strings.

I've updated to use `JSON.parse` instead of the frail `startWith` method.

Reply:
Got it, makes sense!